pkg/commands/workload_apply.go

```diff
@@ -158,20 +153,31 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 		return nil
 	}
 
+	workloadExists := currentWorkload != nil
+
+	if okToPush, err := opts.PublishLocalSource(ctx, c, currentWorkload, workload, shouldPrint); err != nil {
+		return err
+	} else if !okToPush {
+		return nil
+	}
+	// merge annotation only when workload is being created, do not add it when updating workload
+	// since user could be updating another field and annotation must not be added
+	if opts.LocalPath != "" && opts.SourceImage == "" && !workloadExists {
+		workload.MergeAnnotations(apis.LocalSourceProxyAnnotationName, workload.Spec.Source.Image)
+	}
+	// if workload is updated from LSP registry to custom or any other registry through source image,
+	// annotation has to be deleted and workload source image needs to be updated to digested based on opts source image
```

Comment:
We would need similar validation in workload update as well ?

Reply:
I don't think we should add validation for `uptate` command since it's being deprecated. My understanding is that we should not keep on supporting this command